### PR TITLE
Fixes for issue #1683: failed to run accelerate config in colab

### PR DESCRIPTION
**Pull request by Erickrus**
Fixes for issue #1683: failed to run accelerate config in colab

### CODE REVIEW

**Comment by Erickrus** [comment]
How to reformat a file? which tool or standard did you use ?

**Comment by muellerzr** [comment]
@Erickrus please do `pip install -e .[quality]; make style; make quality;`. This will install the right versions of black, ruff, and hf-doc-builder, and run the right quality setup

**Comment by Erickrus** [comment]
For you to decide it can be merged to the base branch @muellerzr

File: src/accelerate/commands/menu/selection_menu.py [hunk 1]
```diff
@@ -15,13 +15,22 @@
...
 import sys
 
 from . import cursor, input
 from .helpers import Direction, clear_line, forceWrite, linebreak, move_cursor, reset_cursor, writeColor
 from .keymap import KEYMAP
 
 
+in_colab = False
+try:
+    in_colab = pkgutil.find_loader("google.colab") is not None
```
Comment (sgugger) [inline src/accelerate/commands/menu/selection_menu.py:29]: Please use [this helper function](https://github.com/huggingface/accelerate/blob/64d7b58c4415a5714c25caa5acb428bec26f94e1/src/accelerate/utils/imports.py#L42) here to be consistent with the rest of the library code.

Comment (Erickrus) [reply src/accelerate/commands/menu/selection_menu.py:29]: > Please use [this helper function](https://github.com/huggingface/accelerate/blob/64d7b58c4415a5714c25caa5acb428bec26f94e1/src/accelerate/utils/imports.py#L42) here to be consistent with the rest of the library code.

Thanks for your advice. I'm looking into it. 
a) Can I use _is_package_available() outside accelerate/utils/imports.py ? 
```python
import builtins
import sys

from ...utils.imports import _is_package_available
from . import cursor, input
from .helpers import Direction, clear_line, forceWrite, linebreak, move_cursor, reset_cursor, writeColor
from .keymap import KEYMAP


in_colab = _is_package_available("google.colab")
```

b) Or I need create another method in imports.py , e.g. 
```
def is_colab_available():
    return _is_package_available("google.colab")
```
puzzled ...

Comment (sgugger) [reply src/accelerate/commands/menu/selection_menu.py:29]: Yes you can do a) :-)

File: src/accelerate/commands/menu/selection_menu.py [hunk 2]
```diff
@@ -15,13 +15,18 @@
...
+import builtins
 import sys
 
+from ...utils.imports import _is_package_available
 from . import cursor, input
 from .helpers import Direction, clear_line, forceWrite, linebreak, move_cursor, reset_cursor, writeColor
 from .keymap import KEYMAP
 
 
+in_colab = _is_package_available("google.colab")
```
Comment (sgugger) [inline src/accelerate/commands/menu/selection_menu.py:27]: Should be maybe be in a try except as it seems to fail instead of returning `False`.

**Review by muellerzr — APPROVED** [review]
Thanks for this! Fantastic! Let's address Sylvain's note here and then we'll be good to merge 😄

File: src/accelerate/commands/menu/selection_menu.py (hunk 2 above)
Comment (Erickrus) [reply src/accelerate/commands/menu/selection_menu.py:27]: Yes, in some environment setup the code is not safe. When I test it in local console and colab, it works. 
I should wrap it with a try except.

**Review by sgugger — APPROVED** [review]
LGTM, thanks for iterating!

**Pull request merged**